app/crud.py

```
@@ -0,0 +1,66 @@
+from .models import ToDo
+
+
+async def create_todo(db: AsyncSession, todo: ToDo):
+    try:
+        db.add(todo)
+        await db.commit()
+        await db.refresh(todo)
+        return todo
+    except Exception as e:
+        await db.rollback()
+        raise HTTPException(status_code=500, detail=f"Error creating ToDo: {e}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)